pygments/lexers/arturo.py

```diff
@@ -0,0 +1,312 @@
+
+        match args:
+            1:open_string,
+            2:exclamation_mark,
+            3:lang_name,
+            4:space_or_newline,
+            5:code,
+            6:close_string
+        """
+        from pygments.lexers import get_lexer_by_name
+
+        # Header's section
+        yield match.start(1),  String.Double  ,  match.group(1)
+        yield match.start(2),  String.Interpol,  match.group(2)
+        yield match.start(3),  String.Interpol,  match.group(3)
+        yield match.start(4),  Text.Whitespace,  match.group(4)
+
+        lexer = None
+        if self.handle_annotateds:
+            try:
+                lexer = get_lexer_by_name( match.group(3).strip() )
+            except ClassNotFound:
+                pass
+        code = match.group(5)
+
+        if lexer is None:
+            yield match.group(5), String, code
+        else:
+            yield from do_insertions([], lexer.get_tokens_unprocessed(code))
+
+        yield match.start(6), String.Double, match.group(6)
+
+
+
+    tokens = {
+        'root': [
+            include('comments'),
+            include('constants'),
+            include('builtin_functions'),
+            include('operators'),
+            (r'.', Name),
+        ],
+
+        'comments': [
+            (r'(;)(.*)$', Comment.Single),
+            include('shebang'),
+        ],
+            'shebang': [
+                (r'^((\s#!)|(#!))(.*)$',
+                        Comment.Hashbang)
+            ],
+
+        'operators': [
+            include('punctuation'),
+            include('sugar'),
+
+            (r'<:|:>|:<|:>'
+             r'|-:|:-'
+             r'|ø|∞'
+             r'|\@|\#|\$|\%|\&|\_|\!|\!\!'
+             r'|\+|\-|\*|\~|\=|\>|\<'
+             r'|==>|<=>|<==>'
+             r'|=>>|<<=>>|<<==>>'
+             r'|-->|<->|<-->'
+             r'|<\\|\<|\>'
+             r'|\=\||\|\='
+             r'|\./|\^|\.\.|\.'
+             r'|\\',
+            Operator)
+
+        ],
+            'punctuation': [
+                (r'[()[\],]',   Punctuation),
+            ],
+            'sugar':[
+                (r'->'
+                 r'|=>'
+                 r'|\|'
+                 r'|::',
+                Name.Decorator),
+            ],
+
+        'constants': [
+            include('boolean'),
+            include('character'),
+            include('color'),
+            include('float'),
+            include('integer'),
+            include('label'),
+            include('literal'),
+            include('strings'),
+            include('type'),
+        ],
+
+            'boolean': [
+                (words(('false', 'true', 'maybe'), suffix=r'\b'),
+                    Name.Constant)
+            ],
+            'character': [
+                (r'`.`', String.Char)
+            ],
+            'color': [
+                (r'#\w+',
+                        Name.Constant)
+            ],
+            'float': [
+                (r'[0-9]+\.[0-9]+',
```

Comment:
both digit parts have to be present? (some languages allow `.5` or `2.`)

Reply:
@birkenfeld Hi! Original Arturo author here!

Floating-point numbers in arturo are always in the form of `(digits+)\.(digits+)` since `.(anything+)` is mostly used to identify `:literal` values: https://arturo-lang.io/documentation/language/#literal

P.S. In theory, it's only a `.` followed by any valid ID characters, but identifying `\.(digits+)` as something else (in that case a floating-point number) would make the whole thing rather confusing 😉

Reply:
Thanks for clarifying!

Reply:
Thank you, @drkameleon!